auto_dev/services/eject/index.py

```diff
@@ -0,0 +1,343 @@
+        try:
+            config = load_autonolas_yaml(component_type, config_path.parent)[0]
+            config["author"] = new_author
+            config["name"] = new_name
+
+            config_file = _get_default_configuration_file_name_from_type(component_type)
+            write_to_file(config_path.parent / config_file, config, file_type=FileType.YAML)
+        except (FileNotFoundError, ValueError) as e:
+            raise ValueError(f"Failed to update component configuration: {e}") from e
+
+    def _run_eject_command(self, component_id: PublicId, component_type: str) -> bool:
+        """Run the aea eject command."""
+        success, _ = self.run_command(
+            f"yes | aea -s eject {component_type} {component_id.author}/{component_id.name}", shell=True
+        )
+        return success
+
+    def _get_and_verify_component_dir(self, component_id: PublicId, component_type: str) -> Path:
+        """Get and verify the component directory exists."""
+        # Check in vendor directory first
+        vendor_dir = Path.cwd() / "vendor" / f"{component_type}s" / component_id.author / component_id.name
+        if vendor_dir.exists():
+            return vendor_dir
+
+        # If not in vendor, check in root directory (where it goes after ejection)
+        root_dir = Path.cwd() / f"{component_type}s" / component_id.name
+        if root_dir.exists():
+            return root_dir
+
+        raise ValueError(f"Component directory not found in vendor ({vendor_dir}) or root ({root_dir})")
+
+    def _fingerprint_component(self, component_id: PublicId, component_type: str) -> bool:
+        """Run fingerprint command for a component."""
+        success, _ = self.run_command(f"aea -s fingerprint {component_type} {component_id.author}/{component_id.name}")
+        return success
+
+    def _update_and_cleanup(self) -> None:
+        """Update agent config and cleanup directories."""
+        agent_config_path = Path.cwd() / DEFAULT_AEA_CONFIG_FILE
+        self.update_agent_config(agent_config_path)
+        self.cleanup_directories()
+
+    def update_agent_config(self, agent_config_path: Path) -> None:
+        """Update agent configuration with new component references."""
+        if not agent_config_path.exists():
+            return
+
+        agent_config = load_autonolas_yaml("agent", agent_config_path.parent)[0]
+        if f"{self.config.component_type}s" not in agent_config:
+            return
+
+        old_component = f"{self.config.fork_id.author}/{self.config.public_id.name}"
+        new_component = f"{self.config.fork_id.author}/{self.config.fork_id.name}"
+
+        agent_config[f"{self.config.component_type}s"] = [
+            x.replace(old_component, new_component).replace(f"{old_component}:", f"{new_component}:")
+            if old_component in x and new_component != old_component
+            else x
+            for x in agent_config[f"{self.config.component_type}s"]
+        ]
+
```

Comment:
lets do this in the overrides as well yeah?

Nice thing is then we will basically get the override ejected for free when we do a component ejection.